api/src/mutations/collection.rs

```diff
@@ -70,13 +71,38 @@ impl Mutation {
             validate_solana_creator_verification(&owner_address, &input.creators)?;
         }
 
+        let TransactionId(credits_deduction_id) = credits
+            .submit_pending_deduction(
+                org_id,
+                user_id,
+                Actions::CreateCollection,
+                input.blockchain.into(),
+                balance,
+            )
+            .await
+            .map_err(|e| match e.kind() {
+                DeductionErrorKind::InsufficientBalance { available, cost } => Error::new(format!(
+                    "insufficient balance: available: {available}, cost: {cost}"
+                )),
+                DeductionErrorKind::MissingItem => Error::new(format!(
+                    "{:?} is not supported by the blockchain {} at this time",
+                    e.item(),
+                    e.blockchain()
+                )),
+                DeductionErrorKind::InvalidCost(_) => Error::new("invalid cost"),
+                DeductionErrorKind::Send(_) => {
+                    Error::new("unable to send credit deduction request")
+                },
+            })?;
```

Comment:
@ray-kast any recommendations how to eliminate the repetition on the map_err. I tried to create a from transform but the LineItem trait on DeductionError was tripping me up.